src/scripts/main.js

```diff
@@ -1,3 +1,264 @@
+          .localeCompare(b.children[1].innerText));
+      break;
+
+    case 'Office':
+      rows.sort((a, b) =>
+        orderSign * (a.children[2].innerText)
+          .localeCompare(b.children[2].innerText));
+      break;
+
+    case 'Age':
+      rows.sort((a, b) =>
+        orderSign * (a.children[3].innerText) - (b.children[3].innerText));
+      break;
+
+    case 'Salary':
+      rows.sort((a, b) =>
+        orderSign * ((+a.children[4].innerText.slice(1).replace(/,/g, ''))
+          - (+b.children[4].innerText.slice(1).replace(/,/g, ''))));
+      break;
+  }
+  rows.forEach(item => tableBody.append(item));
+});
+
+let previousRow;
+
+tableBody.addEventListener('click', (handler) => {
+  if (previousRow) {
+    previousRow.classList.remove('active');
+  }
+
+  const row = handler.target.closest('tr');
+
+  row.classList.add('active');
+  previousRow = row;
+});
+
+table.insertAdjacentHTML('afterend', `
+  <form class="new-employee-form" action="/" method="get">
+    <label>Name:
+      <input id="name" name="name" type="text" data-qa="name">
+    </label>
+    <label>Position:
+      <input
+        id="position"
+        name="position"
+        type="text"
+        data-qa="position"
+      >
+    </label>
+    <label>Office:
+      <select id="office" name="office" data-qa="office">
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="Tokyo">Tokyo</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>Age:
+      <input id="age" name="age" type="number" data-qa="age">
+    </label>
+    <label>Salary:
+      <input id="salary" name="salary" type="number" data-qa="age">
+    </label>
+    <button
+      id="save_button" name="button" type="text" data-qa="salary"
+    >
+      Save to table
+    </button>
+  </for>
+`);
+
+function pushNotification(posTop, posRight, title, description, type) {
+  const body = document.querySelector('body');
+
+  body.insertAdjacentHTML('afterend', `
+    <div class="notification ${type}" data-qa="notification">
+      <h2 class="title">${title}</h2>
+      <p>${description}</p>
+    </div>
+  `);
+
+  const notification = document.querySelector('div');
+
+  notification.style.top = posTop + 'px';
+  notification.style.right = posRight + 'px';
+
+  setTimeout(() => notification.remove(), 5000);
+};
+
+function checkForErrors(employeeName, position, office, age, salary) {
+  const errorMessages = [];
+
+  if (employeeName.length > 0 && employeeName.length < 4) {
+    errorMessages.push('Name should consist of at least 4 letters.');
+  }
+
+  if (age !== '' && (+age < 18 || +age > 90)) {
+    errorMessages.push('Age range should be from 18 to 90.');
+  }
+
+  if (
+    employeeName === ''
```

Comment:
You should `trim` values from string inputs, I can add employee with name that contains only spaces `    `

Reply:
I added trim method for the employeeName.